_source/_posts/2023-03-01-jakarta-ee-oidc.md

```diff
@@ -0,0 +1,497 @@
+
+                JWTVerifier verifier = JWT.require(algorithm)
+                        .withIssuer(openIdConfig.getIssuerUri())
+                        .build();
+
+                jwt = verifier.verify(accessToken);
+                LOGGER.info("JWT decoded. sub=" + jwt.getClaims().get("sub"));
+                request.setAttribute("jwt", jwt);
+
+            } 
+            
+            ...
+
+        }
+
+        chain.doFilter(request, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+}
+```
+
+This code uses Auth0's JWT verifier for Java. Auth0 [has good docs on JWT verification](https://auth0.com/docs/secure/tokens/json-web-tokens/validate-json-web-tokens). If a valid JWT is found and decoded, it is saved in a request attribute and the request is allowed to continue.
+
+Give it a try. Start the project.
+
+```bash
+./mvnw wildfly:run
+```
+
+Make a request to the protected API endpoint (not the OIDC endpoint).
+
+```bash
+http :8080/api/protected
+```
+
+You'll get:
+
+```Bash
+HTTP/1.1 401 Unauthorized
+```
+
+Now, use your OIDC endpoint to retrieve a token. Using a browser, open `http://localhost:8080/protected`.
+
+Authenticate with Auth0. When you are redirected back to the protected servlet page, copy the token value and save it as a variable in a new shell.
+
+```bash
+TOKEN=eyJraWQiOiJqY3dpbGpUcGVZSG1Jajl6ODR3LV...
+```
+
+In that same shell, make a request to the protected API endpoint using the token.
+
+```bash
+http :8080/api/protected "Authorization: Bearer $TOKEN"
+```
+
+It should return something like the following.
+
+```bash
+HTTP/1.1 200 OK
+Connection: keep-alive
+Content-Length: 401
+Content-Type: text;charset=ISO-8859-1
+Date: Tue, 27 Jan 2023 15:02:59 GMT
+
+Welcome, andrew.hughes@mail.com
+...
+
+```
+
+## Keep learning with Jakarta EE and Auth0
+
+You just built a Jakarta Enterprise Edition application that used the new OpenID Connect annotation and implementation built into Jakarta EE 10. You used Auth0 as the OIDC and OAuth 2.0 provider, and you saw how to implement both SSO and JWT authentication.
+
+You can find the source code for this example on GitHub in the [@oktadev/okta-jakartaee-oidc-example](https://github.com/oktadev/okta-jakartaee-oidc-example) repository.
+
+If you liked this post, there's a good chance you'll like similar ones:
+
+- [Introducing Spring Native for JHipster: Serverless Full-Stack Made Easy](/blog/2022/03/03/spring-native-jhipster)
+- [Add Authentication to Your Vanilla JavaScript App in 20 Minutes](/blog/2018/06/05/authentication-vanilla-js)
+- [Mobile Development with Ionic, React Native, and JHipster](/blog/2020/04/27/mobile-development-ionic-react-native-jhipster)
+- [Fast Java Made Easy with Quarkus and JHipster](/blog/2021/03/08/jhipster-quarkus-oidc)
+- [Build a CRUD App with Vue.js, Spring Boot, and Kotlin](/blog/2020/06/26/spring-boot-vue-kotlin)
+- [Add OpenID Connect to Angular Apps Quickly](/blog/2022/02/11/angular-auth0-quickly)
```

Comment:
TODO: Update this list to use more recent posts.

Reply:
@dan-auth0 Do you have a list of popular/related Java blog posts we should add here in the CTA (call to action)?